src/core/util/uri.cc

```diff
@@ -344,6 +362,23 @@ struct QueryParameterFormatter {
   }
 };
 
+struct Authority {
+  std::string user_info;
+  std::string host_port;
+};
+
+// Parse user_info and host_port from authority
+Authority ParseAuthority(absl::string_view authority) {
```

Comment:
I think this function needs to be defined earlier in the file, before where it gets used.

That having been said, once we change the parameters of `URI::Create()` as I suggested elsewhere, then this function will only ever be called from `URI::Parse()`, so we can just inline the code there instead of having a separate function.